rusqlite_migration/build.rs

```diff
@@ -0,0 +1,30 @@
+use std::{
+    env,
+    fs::{read_to_string, File},
+    io::{self, BufWriter, Write},
+};
+
+fn main() {
+    let readme_path = env::var("CARGO_PKG_README").unwrap();
+    println!("cargo:rerun-if-changed={readme_path}");
+
+    let out_dir = env::var("OUT_DIR").unwrap();
+    let readme_for_rustdoc = File::create(format!("{out_dir}/readme_for_rustdoc.md")).unwrap();
+    let mut out = BufWriter::new(readme_for_rustdoc);
+
+    let readme = read_to_string(readme_path).unwrap();
+    readme
+        .lines()
+        .skip_while(|line| line != &"<!-- rustdoc start -->")
+        .skip(1) // Discard the pattern line
+        .try_fold(0, |acc, line| -> Result<usize, io::Error> {
+            Ok(acc + out.write(line.as_bytes())? + out.write(&[b'\n'])?)
```

Comment:
You can probably replace it with `write_fmt`, it would look like this: `Ok(acc + out.write_fmt(format_args!("{line}\n"))?)`

Reply:
Thanks! There is another problem using `.write`, where it can silently fail to write part of the line if it doesn’t fit in the buffer.

In the end, I’ve used the `write!` macro, with a format string as you suggested. It uses write_all under the hood, which will fail if it doesn’t write a line fully.

Reply:
Actually, clippy suggests using `writeln!` in that case :)